File: javer/src/main/java/com/uuhnaut69/javer/repository/RestaurantRepository.java
package com.uuhnaut69.javer.repository;

import com.uuhnaut69.javer.model.Restaurant;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface RestaurantRepository extends JpaRepository<Restaurant, UUID> {

  @EntityGraph(attributePaths = {"dishes"})
  List<Restaurant> findAll();
}

File: kover/src/main/kotlin/com/uuhnaut69/kover/repository/RestaurantRepository.kt
package com.uuhnaut69.kover.repository

import com.uuhnaut69.kover.model.Restaurant
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface RestaurantRepository : JpaRepository<Restaurant, UUID> {

    @EntityGraph(attributePaths = ["dishes"])
    override fun findAll(): List<Restaurant>
}